envoyextensions/extensioncommon/basic_envoy_extender.go

```diff
@@ -57,6 +61,23 @@ type BasicExtension interface {
 	// Then PatchClusters is called for each individual cluster.
 	PatchClusters(*RuntimeConfig, ClusterMap) (ClusterMap, error)
 
+	// PatchClusterLoadAssignment patches a cluster load assignment to include the custom Envoy configuration
+	// required to integrate with the built in extension template.
+	PatchClusterLoadAssignment(*RuntimeConfig, *envoy_endpoint_v3.ClusterLoadAssignment) (*envoy_endpoint_v3.ClusterLoadAssignment, bool, error)
```

Comment:
We don't have a use case for list-order operations on this resource, so only singular patch op is supported